CONTRIBUTING.md

```diff
@@ -0,0 +1,194 @@
+
+**Describe the solution you'd like**
+A clear and concise description of what you want to happen.
+
+**Is it blocking?**
+Is this issue blocking any of your work? If it is blocking any open source project, you can share the link of the issue
+
+**Describe alternatives you've considered**
+A clear and concise description of any alternative solutions or features you've considered.
+
+**Additional context**
+Add any other context or screenshots about the feature request here.
+
+**Other**
+Other information like relevant issues, external links, etc
+```
+
+[Feature Request Link]: https://github.com/uuid-rs/uuid/issues/new?template=Feature_request.md
+
+# Bug Reports
+[Bug Reports]: #bug-reports
+
+While no one likes bugs, they are an unfortunate reality in software. Remember
+we can't fix bugs we don't know about, so don't be shy about reporting.
+
+If you have the chance, please [search existing issues], as there is a chance
+that someone has already reported your error. This isn't strictly needed, as
+sometimes you might not what exactly you are looking for.
+
+File your issue with a descriptive title, as this helps others find your issue.
+
+Reporting a bug is as easy as following [this link][Bug Report Link] and
+filling it in. The template used is shown below:
+
+```markdown
+**Describe the bug**
+A clear and concise description of what the bug is.
+
+**To Reproduce**
+Steps to reproduce the behavior:
+1. ...
+2. ...
+3. ...
+
+**Expected behavior**
+A clear and concise description of what you expected to happen.
+
+**Screenshots**
+If applicable, add screenshots to help explain your problem.
+
+**Specifications (please complete the following information):**
+- Target
+- Version [e.g. 1.0]
+- Features Enabled
+
+**Additional context**
+Add any other context about the problem here.
+
+**Other**
+Other information like relevant issues, external links, etc
+```
+
+Sometimes a backtrace may be needed. In that case, set `RUST_BACKTRACE`
+environment variable to `1`. For example:
+
+```bash
+$ RUST_BACKTRACE=1 cargo build
+```
+
+**Aside** We welcome pull requests for your own bug reports.
+
+[Bug Report Link]: https://github.com/uuid-rs/uuid/issues/new?template=Bug_report.md
+
+# Pull Requests
+[Pull Requests]: #pull-requests
+
+Pull requests are the primary mechanism we use to change Uuid. GitHub itself
+has some [great documentation] on using the Pull Request feature. We use the
+"fork and pull" model described [here][fnp], where contributors push changes to
+their personal fork and create pull requests to bring those changes into the
+source repository.
+
+Unless the changes are fairly minor (like documentation changes or tiny
+patches), we require pull requests to relevant issues.
+
+Please make pull requests against:
+* `master` when making non-breaking changes 
+* `breaking` if your changes alter the public API in a breaking manner
+
+*Note* Our minimum rust version supported currently is `1.18.0`. Make sure you
```

Comment:
Do we want to list the minimum version in here? It'll be easy to forget to update it. Maybe we should just point to the `travis.yaml` where the minimum version is listed?